add ref to mounting emptyDir for ephemeral storage (#1404)
astronomer/docs: astro/kubernetespodoperator.md

Applying the code above ensures that when this DAG runs, it will launch a Kubernetes Pod with exactly 800m of CPU and 3Gi of memory as long as that infrastructure is available in your cluster. Once the task finishes, the Pod will terminate gracefully.

### Mount a temporary directory (_AWS only_)

> Astronomer provisions `m5d` workers with NVMe SSD volumes. These volumes can be used by tasks for storage which persists as long as the task is running. To run a task run the KubernetesPodOperator that utilizes temporary storage:

Reviewer: Or m6id https://docs.astronomer.io/astro/resource-reference-aws#worker-node-types

1. Create a [worker queue](configure-worker-queues.md) with `m5d` workers. See [Modify a cluster](modify-cluster.md) for instructions on adding `m5d` workers to your cluster.
2. Mount and [emptyDir volume](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir-configuration-example) to the KubernetesPodOperator like in the following example:

    ```python{5-14,26-27}
    from airflow.configuration import conf
    from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
    from kubernetes.client import models as k8s

    volume = k8s.V1Volume(
        name="cache-volume",
        emptyDir={},
    )

    volume_mounts = [
        k8s.V1VolumeMount(
            mount_path="/cache", name="cache-volume"
        )
    ]

    example_volume_test = KubernetesPodOperator(
    namespace=namespace,
    image="<your-docker-image>",
    cmds=["<commands-for-image>"],
    arguments=["<arguments-for-image>"],
    labels={"<pod-label>": "<label-name>"},
    name="<pod-name>",
    resources=compute_resources,
    task_id="<task-name>",
    get_logs=True,
    volume_mounts=volume_mounts,
    volumes=[volume],
    )
    ```
    

>     Your task can now utilize the configured volume as storage for simple operations. However, Astronomer recommends setting Airflow connection to an external storage service such as Amazon S3 for larger operations requiring storage. See [Passing data between Airflow tasks](https://www.docs.astronomer.io/learn/airflow-passing-data-between-tasks/).

Reviewer 2: ```suggestion
    Your task can now write data to the configured volume. Astronomer recommends using this volume only with small amounts of data. For larger operations requiring storage, Astronomer recommends setting an Airflow connection to an external storage service such as Amazon S3 . See [Passing data between Airflow tasks](https://www.docs.astronomer.io/learn/airflow-passing-data-between-tasks/).
```

>     Your task can now write data to the configured volume. Astronomer recommends using this volume only with small amounts of data. For larger operations requiring storage, Astronomer recommends setting an Airflow connection to an external storage service such as Amazon S3 . See [Passing data between Airflow tasks](https://www.docs.astronomer.io/learn/airflow-passing-data-between-tasks/).

Reviewer: Can you point to the resources table telling them how much space is available? "small amounts of data" is too vague.

Reviewer: There is 20GB capacity for all instance types except for the m5d and m6id types - we need to tell them how much space is available according to https://aws.amazon.com/ec2/instance-types/m6i/ & https://aws.amazon.com/ec2/instance-types/m5/

Reviewer: We don't necessarily need to give them numbers - we can just link the above

>     Your task can now write data to the configured volume. Astronomer recommends using this volume only with small amounts of data. For larger operations requiring storage, Astronomer recommends setting an Airflow connection to an external storage service such as Amazon S3 . See [Passing data between Airflow tasks](https://www.docs.astronomer.io/learn/airflow-passing-data-between-tasks/).

Reviewer: ```suggestion
    Your task can now write data to the configured volume. Astronomer recommends using this volume only with small amounts of data. For larger operations requiring storage, Astronomer recommends setting an Airflow connection to an external storage service such as Amazon S3, see [Passing data between Airflow tasks](https://www.docs.astronomer.io/learn/airflow-passing-data-between-tasks/).
```

1. Create a [worker queue](configure-worker-queues.md) with `m5d` workers. See [Modify a cluster](modify-cluster.md) for instructions on adding `m5d` workers to your cluster.
2. Mount and [emptyDir volume](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir-configuration-example) to the KubernetesPodOperator like in the following example:

    ```python{5-14,26-27}
    from airflow.configuration import conf
    from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
    from kubernetes.client import models as k8s

    volume = k8s.V1Volume(
        name="cache-volume",
        emptyDir={},
    )

    volume_mounts = [
        k8s.V1VolumeMount(
            mount_path="/cache", name="cache-volume"
        )
    ]

    example_volume_test = KubernetesPodOperator(
    namespace=namespace,
    image="<your-docker-image>",
    cmds=["<commands-for-image>"],
    arguments=["<arguments-for-image>"],
    labels={"<pod-label>": "<label-name>"},
    name="<pod-name>",
    resources=compute_resources,
    task_id="<task-name>",
    get_logs=True,
    volume_mounts=volume_mounts,
    volumes=[volume],
    )
    ```

>     
>     Your task can now write data to the configured volume.

Reviewer 2: ```suggestion
```

>     Your task can now write data to the configured volume. Astronomer recommends that your tasks use no more storage than 20GB divided by your configured maximum tasks per worker. See [Worker autoscaling logic](configure-worker-queues.md#worker-autoscaling-logic). For larger operations requiring storage, Astronomer recommends setting an Airflow connection to an external storage service such as Amazon S3. See [Passing data between Airflow tasks](https://www.docs.astronomer.io/learn/airflow-passing-data-between-tasks/).

Author: I don't think we should be recommending 20GB here unless we have language excepting the NVME based instances. Possibly better to just link to the AWS docs about instance storage.
